hadoop-tools/hadoop-azure/dev-support/testrun-scripts/testsupport.sh

```diff
@@ -0,0 +1,146 @@
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+conffile=src/test/resources/azure-auth-keys.xml
+bkpconffile=src/test/resources/azure-auth-keys_BKP.xml
+logdir=target/testlogs
+testresultsregex="Results:(\n|.)*?Tests run:"
+testresultsfilename=
+starttime=
+threadcount=
+defaultthreadcount=8
+
+validate() {
+  if ! [ -s "$conffile" ] ; then
+    echo "Exiting. conf file is empty"
+    exit -1
+  fi
+  if ! xmlstarlet val --well-formed "$conffile" ; then
+    echo "Exiting. conf file is not well formed xml"
+    exit -1
+  fi
+  if [ -z "$threadcount" ] ; then
+    threadcount=$defaultthreadcount
+  fi
+  numberegex='^[0-9]+$'
+  if ! [[ $threadcount =~ $numberegex ]] ; then
+    echo "Exiting. The script param (threadcount) should be a number"
+    exit -1
+  fi
+  if [ -z "$scenario" ]; then
+   echo "Exiting. scenario cannot be empty"
+   exit -1
+  fi
+  propertiessize=${#properties[@]}
+  valuessize=${#values[@]}
+  if [ "$propertiessize" -lt 1 ] || [ "$valuessize" -lt 1 ] || [ "$propertiessize" -ne "$valuessize" ]; then
+    echo "Exiting. Both properties and values arrays has to be populated and of same size. Please check for scenario $scenario"
+    exit -1
+  fi
+
+  if ! [ -f "$conffile" ]; then
+    echo "Exiting. $conffile missing"
+    exit -1
+  fi
+}
+
+checkdependencies() {
+  if ! [ "$(command -v pcregrep)" ]; then
+    echo "Exiting. pcregrep is required to run the script."
+    exit -1
+  fi
+  if ! [ "$(command -v xmlstarlet)" ]; then
+    echo "Exiting. xmlstarlet is required to run the script."
+    exit -1
+  fi
+}
+
+changeconf() {
+  xmlstarlet ed -P -L -d "/configuration/property[name='$1']" $conffile
+  xmlstarlet ed -P -L -s /configuration -t elem -n propertyTMP -v "" -s /configuration/propertyTMP -t elem -n name -v "$1" -r /configuration/propertyTMP -v property $conffile
+  if ! xmlstarlet ed -P -L -s "/configuration/property[name='$1']" -t elem -n value -v "$2" $conffile
+  then
+    echo "Exiting. Changing config property failed."
+    exit -1
+  fi
+}
+
+removeproperties() {
+  propertiessize=${#properties[@]}
+  for ((i = 0; i < propertiessize; i++)); do
+    key=${properties[$i]}
+    xmlstarlet ed -P -L -d "/configuration/property[name='$key']" $conffile
+  done
+}
+
+testwithconfs() {
+  propertiessize=${#properties[@]}
+  valuessize=${#values[@]}
+  if [ "$propertiessize" -ne "$valuessize" ]; then
+    echo "Exiting. Number of properties and values differ for $scenario"
+    exit -1
+  fi
+  for ((i = 0; i < propertiessize; i++)); do
+    key=${properties[$i]}
+    val=${values[$i]}
+    changeconf "$key" "$val"
+  done
+  mvn -T 1C -Dparallel-tests=abfs -Dscale -DtestsThreadCount=$threadcount verify >> "$testlogfilename"
```

Comment:
only one scenario needs to run mvn test. Maybe set -Dtest=undefined and so they get skipped

At the very least, let the caller declare a set of options to go down to maven